### PR TITLE
feat(replays): Emit has hydration error analytic

### LINKED ISSUE #61847: Add a hydration breadcrumb presence analytic event
We're trying to find examples of replays that were viewed and would've had the hydration breadcrumb.

ACs:
When they do not have the feature hydration modal feature flag, emit an analytic event if the hydration error breadcrumb is present in the replay.

### PR DESCRIPTION
**Pull request by scttcper**
fixes #61847

### CODE REVIEW

File: static/app/utils/replays/hooks/useLogReplayDataLoaded.tsx [hunk 1]
```diff
@@ -45,6 +45,23 @@ function useLogReplayDataLoaded({fetchError, fetching, projectSlug, replay}: Pro
       finished_at_delta: replay.timestampDeltas.finishedAtDelta,
       replay_id: replayRecord.id,
     });
+
+    // TODO(session-replay-show-hydration-errors): remove this analytic on GA
+    if (!organization.features.includes('session-replay-show-hydration-errors')) {
+      // @ts-expect-error the hydration frames are filtered when the flag is off
+      const allFrames = replay._sortedBreadcrumbFrames;
```
Comment (ryan953) [inline static/app/utils/replays/hooks/useLogReplayDataLoaded.tsx:52]: This is meant to be a private field, so a) i'm surprised typescript isn't complaining b) can you call `getChapterFrames` instead

Comment (scttcper) [reply static/app/utils/replays/hooks/useLogReplayDataLoaded.tsx:52]: getChapterFrames is filtering out the frames i want https://github.com/getsentry/sentry/blob/615d9f558639b13cae5f023d463aeee152bc2875/static/app/utils/replays/replayReader.tsx#L298

Comment (ryan953) [reply static/app/utils/replays/hooks/useLogReplayDataLoaded.tsx:52]: oh right, the other feature.

why is TS not complaining though?

Comment (scttcper) [reply static/app/utils/replays/hooks/useLogReplayDataLoaded.tsx:52]: ts-expect-error?

Comment (ryan953) [reply static/app/utils/replays/hooks/useLogReplayDataLoaded.tsx:52]: what if the ReplayReader class had a public method 'has hydration crumb'

something like:
`replay._sortedBreadcrumbFrames.some(...);`

**Review by ryan953 — APPROVED** [review]

**Pull request merged**